### PR TITLE
cmake: 3.27.8 -> 3.28.2

### PR DESCRIPTION
**Pull request by tfc**
## Description of changes

This PR updates cmake to 3.28.2

I had to change the patch 005 as it did not apply any longer.
Also, i had to add a new dependency on macOS.

I can imagine that this needs to be tested extensively because this package has a lot of dependendants.
Not sure how to do that in the best way, what is normally done when touching this package?

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by tfc** [comment]
Pinging the maintainers @ttuegel @lnl7 @AndersonTorres

**Comment by AndersonTorres** [comment]
Duplicate of https://github.com/NixOS/nixpkgs/pull/275284

**Comment by tfc** [comment]
> Duplicate of #275284

Thank you for pointing me to that PR. I looked for cmake PRs but the search returned too many so i didn't find the needle in the haystack.

However, it is not a duplicate as it does not upgrade to 3.28.1 but 3.28.2 instead.

As @gracicot mentioned in the other PR, it may increase compatibility.

**Comment by raboof** [comment]
I have no preference whether we merge #275284 first and then rebase this one, or go straight for this one (assuming they are on-par in terms of changes). On any case, though, this one should target `staging` rather than `master` (https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#changes-causing-mass-rebuilds)

**Comment by tfc** [comment]
@raboof I rebased the PR to `staging` and made sure that cmake still builds there.

**Comment by AndersonTorres** [comment]
> However, it is not a duplicate as it does not upgrade to 3.28.1 but 3.28.2 instead.

Duplicate work, nonetheless.
I would suggest a "hard merge":

1. Bump 3.27.8 -> 3.28.1 first, giving credits to @gracicot for the hash update;
2. Bump 3.28.1 -> 3.28.2

**Comment by tfc** [comment]
@gracicot what is your opinion on that?

**Comment by gracicot** [comment]
@tfc I would okay with either way, as long as the package gets upgraded smoothly. Following what @AndersonTorres seems sane and would still result in CMake getting upgraded to the latest version, and probably make 3.28 get to users faster. I'll probably go and rebase my PR just to make it more mergeable, then once merged I'll let you continue here.

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/by-name/cm/cmake/005-remove-systemconfiguration-dep.diff
Comment (raboof) [inline pkgs/by-name/cm/cmake/005-remove-systemconfiguration-dep.diff:26]: Being closer to upstream again seems like an improvement, but since I didn't see any comment about this in the thread I wonder if this change was intentional?

Comment (tfc) [reply pkgs/by-name/cm/cmake/005-remove-systemconfiguration-dep.diff:26]: You're right, it was not intentional. Let me remove that line in the patch again.

**Review by raboof — APPROVED** [review]

**Comment by Lassulus** [comment]
@GrahamcOfBorg eval

**Pull request merged**
